actions/arteria_delivery_service.py

```diff
@@ -86,21 +87,48 @@ class ArteriaDeliveryQuerier(ArteriaQuerierBase):
     delivery_in_progress = 'delivery_in_progress'
     delivery_successful = 'delivery_successful'
     delivery_failed = 'delivery_failed'
+    delivery_skipped = 'delivery_skipped'
+
+    valid_states = [pending, mover_processing_delivery, mover_failed_delivery,
+                    delivery_in_progress, delivery_successful, delivery_failed]
+
+    successful_states = delivery_successful
 
     def __init__(self, logger):
         super(ArteriaDeliveryQuerier, self).__init__(logger)
 
     def successful_status(self):
-        return self.delivery_successful
+        return self.successful_states
 
     def failed_status(self):
         return [self.mover_failed_delivery, self.delivery_failed]
 
     def valid_status(self):
-        valid_states = [self.pending, self.mover_processing_delivery, self.mover_failed_delivery,
-                        self.delivery_in_progress, self.delivery_successful, self.delivery_failed]
-        return valid_states
-
+        return self.valid_states
+
+    def query_for_status(self, link, skip_mover):
+        if skip_mover:
+            self.valid_states += self.delivery_skipped
+            self.successful_states = self.delivery_skipped
+
+        while True:
+            response = requests.get(link)
+            response_as_json = json.loads(response.content)
+            status = response_as_json["status"]
+
+            if status == self.successful_status():
```

Comment:
Perhaps you could rename `successful_states -> successful_state`, as I was first led to believe that you might want to re-define it as a list with more entries later in the code. But I guess from this that you only allow one state per query.